### DISCUSSION TITLE
Question on use of `~/` in paths

### DISCUSSION THREAD

**Discussion by wwahammy** [ElMassimo/js_from_routes#28] [General]
First off, thanks so much @ElMassimo  for this library. I like that it helps without making all too many assumptions.

I had a question about the decision to use paths that start with `~/` in some of the templates. For example, in the [template_all.js.erb]:
https://github.com/ElMassimo/js_from_routes/blob/86002a920bb6ab4be376957f8f23887011fd9c58/js_from_routes/lib/js_from_routes/template_all.js.erb#L4

Why would one use that instead of using `./` there? That makes more sense to me immediately but I've never used `~/` so it's possible I'm overlooking something.

**Comment by ElMassimo** [comment]
Hi Eric!

The `~/` in the default template is referring to an [import alias](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89), commonly configured in Webpack and Vite projects to point to the source code root.

For example, [Vite Ruby](https://vite-ruby.netlify.app/) configures it by default, so this would work out of the box.

The main advantage of using an alias, is that absolute paths work everywhere. Relative paths could make the template invalid under some circumstances, for example, controllers with namespaces that result in files placed in nested directories.

You [can use a custom template](https://js-from-routes.netlify.app/guide/codegen.html#using-a-different-template) if you prefer using a relative path, or don't want to define an import alias (or if you are using a bundler that doesn't support aliases).
